src/rdf-repository/grafter/rdf/sparql.clj

```diff
@@ -3,27 +3,45 @@
             [grafter.rdf.repository :refer [repo sparql-repo ->connection]]
             [grafter.rdf.repository :as repo]
             [grafter.rdf.io :refer [->sesame-rdf-type]]
-            [clojure.java.io :refer [resource]]))
+            [clojure.java.io :refer [resource]]
+            [clojure.string :as str]))
+
+(defn pre-process-limit-clauses
+  "look through a query string and replaces limit clauses with values supplied
+  as maps against matching SPARQL ?variable names or a limit integer"
+  [sparl-query limits]
+  (reduce (fn [memo [key val]]
+            (if-let [pattern (cond
+                               (integer? key)
+                               (str "(?i)LIMIT\\s+" key)
+
+                               (or (string? key) (keyword? key))
```

Comment:
Wondering whether this check is needed.

If it's not a `string?` or a `keyword?` it looks like this code will possibly fail because the `cond` returns `nil`.  Perhaps these should be replaced with an `:else (str ,,,regex,,, (name key)` which will at least always raise an error inside `name`...  I suppose if we did that though an `if` with an else clause would be enough.

Alternatively we could add a more explicit error to `:else` e.g. `:else (throw (ex-info "Pass us a string or a keyword or an integer" {}))` though this might be going too far... especially as we may add `clojure.spec`s to this stuff in the future.